citizennet/purescript-ocelot
# Consolidate Typeahead Modules

## What does this pull request do?

There are two changes here: 
1. Removing `Component` from the Typeahead module namespace.
2. Consolidating the Typeahead modules into one. 

The commit messages provide more context.

Assigning this specifically to @boygao1992 as this will have the biggest impact on the Wildcat codebase. However, a couple `sed` commands are enough to fix 99% of the errors, leaving just a handful that require manual attention.

## How should this be manually tested?

If you don't want to rely on CI, you can:
- [ ] Run `make build-ui` or `make ui-guide` and verify the UI Guide still builds and displays as expected.

If you want to go a step further and validate these changes against Wildcat, you can:

1. In Wildcat, edit `packages.dhall`:
    ```diff
     -  { ocelot
     + { ocelot = ../purescript-ocelot/spago.dhall as Location
    ```
    This tells spago to use your local checkout of Ocelot.
1. Run these `sed` commands to rename the relevant module imports:
    ```sh
    find web/purs/src -type f -name '*.purs' | xargs sed -i 's/Ocelot.Component.Typeahead/Ocelot.Typeahead/g'
    find web/purs/src -type f -name '*.purs' | xargs sed -i 's/Ocelot.Typeahead.Render/Ocelot.Typeahead/g'
    I had to play around with a few variants of those before I found one that covered all instances properly, so you may have to tweak those.
1. Lastly, if you run `make` you should only be left with a few edge cases that you have to address manually.
1. [ ] After following the above steps, you should be left with a successful build.
